cmd/common/dymint_config.go

```diff
@@ -0,0 +1,274 @@
+package common
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/server"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	tmos "github.com/tendermint/tendermint/libs/os"
+
+	dymintconf "github.com/dymensionxyz/dymint/config"
+	dymintconv "github.com/dymensionxyz/dymint/conv"
+)
+
+// DymintContextKey defines the context key used to retrieve a server.Context from
+// a command's Context.
+const (
+	DymintContextKey = sdk.ContextKey("dymint.context")
+)
+
+type DymintContext struct {
+	Viper  *viper.Viper
+	Config *dymintconf.NodeConfig
+}
+
+func NewDefaultContext() *DymintContext {
+	return NewContext(viper.New(), &dymintconf.DefaultNodeConfig)
+}
+
+func NewContext(v *viper.Viper, config *dymintconf.NodeConfig) *DymintContext {
+	return &DymintContext{v, config}
+}
+
+// GetDymintContextFromCmd returns a Context from a command or an empty Context
+// if it has not been set.
+func GetDymintContextFromCmd(cmd *cobra.Command) *DymintContext {
+	if v := cmd.Context().Value(DymintContextKey); v != nil {
+		dymintCtxPtr := v.(*DymintContext)
+		return dymintCtxPtr
+	}
+
+	return NewDefaultContext()
+}
+
+// SetCmdDymintContext sets a command's Context value to the provided argument.
+func SetCmdDymintContext(cmd *cobra.Command, dymintCtx *DymintContext) error {
+
+	v := context.WithValue(cmd.Context(), DymintContextKey, dymintCtx)
+	cmd.SetContext(v)
+
+	return nil
+}
+
+func DymintConfigPreRunHandler(cmd *cobra.Command) error {
+	dymintCtx := NewDefaultContext()
+	clientCtx, err := client.GetClientQueryContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	// Bind command-line flags to Viper
+	if err := dymintCtx.Viper.BindPFlags(cmd.Flags()); err != nil {
+		return err
+	}
+	if err := dymintCtx.Viper.BindPFlags(cmd.PersistentFlags()); err != nil {
+		return err
+	}
+
+	rootDir := dymintCtx.Viper.GetString(flags.FlagHome)
+	configPath := filepath.Join(rootDir, "config")
+	dymintCfgFile := filepath.Join(configPath, "dymint.toml")
+
+	//FIXME: bind Dymint flags as well to allow overriding config file values
+
+	//prepare default settlement config
+	//TODO: wrap in method
+	dymintCtx.Config.SettlementConfig.RollappID = clientCtx.ChainID
+	if dymintCtx.Config.SettlementLayer == "mock" {
+		dymintCtx.Config.SettlementConfig.KeyringHomeDir = rootDir
+	} else {
+		dymintCtx.Config.SettlementConfig.KeyringHomeDir = rootify("/sequencer", rootDir)
+	}
+
+	dymintCtx.Viper.SetConfigType("toml")
+	dymintCtx.Viper.SetConfigName("dymint")
+	dymintCtx.Viper.AddConfigPath(configPath)
+	dymintCtx.Viper.SetEnvPrefix("DYMINT")
+	dymintCtx.Viper.AutomaticEnv()
+
+	_, err = os.Stat(dymintCfgFile)
```

Comment:
Seems like you are checking twice if the file exists. Here and in line 147 after calling the function.